### PR TITLE
Implement C++17 filesystem

### PR DESCRIPTION
**Pull request by feliwir**
Requires updating Workflow Version (happening in another PR already)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -78,6 +80,7 @@ endif()
 # This doesn't really work yet, work ongoing to make it usable
 option(STANDALONE "Build a standalone version." ${DEFAULT_STANDALONE})
 option(USE_GAMEMATH "Use own maths library rather than libc version for this platform." ON)
+option(USE_STDFS "Use C++ 17 filesystem for crossplatform file handling." ${DEFAULT_STDFS})
```
Comment (xezon) [inline CMakeLists.txt]: Maybe call this `USE_STDFILESYTEM` and `BUILD_WITH_STDFILESYSTEM` accordingly.

File: src/platform/stdlocalfilesystem.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+bool StdLocalFileSystem::Create_Directory(Utf8String dir_path)
+{
+    if (dir_path.Is_Empty()) {
+        return false;
+    }
+
+    std::error_code ec;
+    bool result = fs::create_directory(dir_path.Str(), ec);
```
Comment (xezon) [inline src/platform/stdlocalfilesystem.cpp:129]: `bool created`

**Review by xezon — DISMISSED** [review]
Looks good.

**Comment by duncanspumpkin** [comment]
Shouldn't this just be tied to if building with stl_port or not?

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.19.0)
+cmake_minimum_required(VERSION 3.18.0)
```
Comment (xezon) [inline CMakeLists.txt]: Why is this changed?

**Review by xezon — APPROVED** [review]
